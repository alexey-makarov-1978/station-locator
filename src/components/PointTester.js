import React from "react";
import PropTypes from "prop-types";

import SuitableStation from "./StationLocator";

function PointTester({ point }) {
  // component checks suitable location for the passed point
  // if power = 0 station object doesnt contain data (null)
  const { station, power } = SuitableStation(point);

  return (
    <div>
      {power > 0
        ? `Best link station for point ${point.X}, ${point.Y} is ${
            station.Location.X
          }, ${station.Location.Y} with power ${power.toFixed(2)}`
        : `No link station within reach for point ${point.X}, ${point.Y}`}
    </div>
  );
}

PointTester.propTypes = {
  point: PropTypes.object.isRequired,
};

export default PointTester;
