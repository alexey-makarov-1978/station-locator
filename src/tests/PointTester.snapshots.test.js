import React from "react";
import PointTester from "../components/PointTester";
import renderer from "react-test-renderer";
import Point from "../components/Point";

const point1 = new Point(100, 100);
const point2 = new Point(0, 0);

it("PointTester renders no matching message for 100, 100", () => {
  const tree = renderer.create(<PointTester point={point1} />);
  expect(tree).toMatchSnapshot();
});

it("PointTester renders matching message for 0, 0", () => {
  const tree = renderer.create(<PointTester point={point2} />);
  expect(tree).toMatchSnapshot();
});
