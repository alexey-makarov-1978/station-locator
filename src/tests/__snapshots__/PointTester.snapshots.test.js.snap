// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointTester renders matching message for 0, 0 1`] = `
<div>
  Best link station for point 0, 0 is 0, 0 with power 100.00
</div>
`;

exports[`PointTester renders no matching message for 100, 100 1`] = `
<div>
  No link station within reach for point 100, 100
</div>
`;
